### PR TITLE
#167542716 Add all API endpoints 

### PR DESCRIPTION
**Pull request by StevenDerrick**
#### What does this PR do?
It will merge all api endpoints and add admin login functionality to the UI

#### Description of Tasks to be completed?

-create user post request signup endpoint
 -create user post request sign-in endpoint
 -create get request all trips and by ID endpoint
 -create admin post request to create a trip endpoint
 -create admin patch request cancel trip endpoint
 -create user post request to booking a sit endpoint
 -create user get a request to fetch all bookings endpoint
 -create user delete request to delete a booking endpoint
 -test few get request using mocha
 -add some few features of admin login in the UI basing on the feedback

#### How should this be manually tested?
After merging this PR, You can view the changes on the website hosted on gh-pages at
https://stevenderrick.github.io/Way-Farer and endpoints can be consumed by postman

#### What are the relevant pivotal tracker stories?
-https://www.pivotaltracker.com/story/show/167542716
-https://www.pivotaltracker.com/story/show/167755890
-https://www.pivotaltracker.com/story/show/167763703
-https://www.pivotaltracker.com/story/show/167763833
-https://www.pivotaltracker.com/story/show/167763926
-https://www.pivotaltracker.com/story/show/167764160
-https://www.pivotaltracker.com/story/show/167764259
-https://www.pivotaltracker.com/story/show/167764444
-https://www.pivotaltracker.com/story/show/167764562

### CODE REVIEW

File: server/controllers/user.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      last_name: req.body.last_name,
+      email: req.body.email,
+      password: hash,
+      is_admin: false,
+    };
+    users.push(newUser);
+    res.status(201).json({
+      status: 201,
+      data: {
+        id: newUser.id,
```
Comment (Oluwasegun-AA) [inline server/controllers/user.js]: @StevenDerrick , you should only return a token in the data key here

File: server/controllers/user.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+exports.users_sign_up = (req, res, next) => {
+  const user = users.find(c => c.email === req.body.email);
+  if (user) {
+    return res.status(401).json({
+      status: 401,
+      error: 'Invalid email or password',
+    });
+  }
+  const schema = {
+    first_name: Joi.string().min(3).required(),
```
Comment (Oluwasegun-AA) [inline server/controllers/user.js]: @StevenDerrick , you should not validate your payload in the controller. this should be done in the middlewares

File: server/controllers/user.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        },
+        process.env.JWT_KEY,
+        {
+          expiresIn: '1h',
+        },
+      );
+      return res.status(200).json({
+        status: 200,
+        data: {
+          token,
```
Comment (Oluwasegun-AA) [inline server/controllers/user.js]: @StevenDerrick , note  you are returning just the token and nothing else

File: server/controllers/user.js (hunk 1 above)
Comment (StevenDerrick) [reply server/controllers/user.js]: I going to fix it

File: server/controllers/user.js (hunk 2 above)
Comment (StevenDerrick) [reply server/controllers/user.js]: I am going to fix it

File: server/controllers/user.js (hunk 3 above)
Comment (StevenDerrick) [reply server/controllers/user.js]: Yes

**Pull request merged**